src/Orleans.Core.Abstractions/IDs/ActivationId.cs

```diff
@@ -40,31 +34,24 @@ private ActivationId(UniqueKey key)
 
         public static ActivationId NewId()
         {
-            return FindOrCreate(UniqueKey.NewKey());
+            return GetActivationId(UniqueKey.NewKey());
         }
 
         // No need to encode SiloAddress in the activation address for system target. 
         // System targets have unique grain ids and addressed to a concrete silo, so in fact we don't need ActivationId at all for System targets.
         // Need to remove it all together. For now, just use grain id as activation id.
         public static ActivationId GetDeterministic(GrainId grain)
         {
-            return interner.FindOrCreate(grain, CreateActivationId);
-
-            static ActivationId CreateActivationId(GrainId grainId)
```

Comment:
I thought that the compiler was able to optimize this use, but I suppose not?

Reply:
I just deleted these lines for less code, there is no perf benefit from that and it would get inlined either way.

Reply:
The compiler allocates a new delegate each time if it doesn't use the lambda syntax, even for static methods. [SharpLab sample](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAGAAkIEYA6AYQl1xjA2wgDsBnW9sAVylhsMAbgCwAKAmEAzOQpJyAJlI1SAbwmktchbACGAE3a4Anip79BGACLZmrNnqgmAPNiFp3GAHykvMKDYA4W1STW1w0K0ZHT8hUgBxGBtkgIBbd2wOFjAACi9SAHMoPXcASkiojXEo2vIAdjiMAKCoKiSMAHkoAEEDA1zi0rY0FX1mnvsANz0WdgBJAzKxGrrQygUCmnGYSZYZubZF/Pih90WygF5vFEUVtYBfSsr12Q2mxOTrVKgMtiycooThgiiVyi9tNU1utGv5AgF2slun0BmcRn5rrdlhCtE9xA8gA=).

Reply:
That's a shame - I had hope they'd "fix" this for static local functions. IIRC, it's either part of the language spec or they didn't seem it worthy of optimizing yet.